packages/web/src/pages/upload-page/UploadPageNew.tsx

```diff
@@ -1,23 +1,159 @@
+  const openMultiTrackNotification = useSelector(
+    // TODO: move to selectors.ts
+    (state: CommonState) => state.upload.openMultiTrackNotification
+  )
+  const playing = useSelector(getPlaying)
+
+  const [tracks, setTracks] = useState<TrackForUpload[]>([])
+  const [uploadTrackError, setUploadTrackError] =
+    useState<Nullable<ErrorType>>(null)
+  // TODO: support upload types when directing to next page
+  const [uploadType, setUploadType] = useState<UploadType>(
+    UploadType.INDIVIDUAL_TRACK
+  )
+  const [previewIndex, setPreviewIndex] = useState(-1)
+  const [preview, setPreview] = useState<Nullable<HTMLAudioElement>>()
+
+  const onSelectTracks = useCallback(
+    async (selectedFiles: File[]) => {
+      const existing = new Set(
+        tracks.map(
+          ({ file }: { file: File }) => `${file.name}-${file.lastModified}`
+        )
+      )
+      selectedFiles = selectedFiles.filter(({ name, lastModified }) => {
+        const id = `${name}-${lastModified}`
+        if (existing.has(id)) return false
+        existing.add(id)
+        return true
+      })
+
+      const processedFiles = processFiles(selectedFiles, (_name, reason) =>
+        setUploadTrackError({ reason })
+      )
+      const processedTracks = (await Promise.all(processedFiles)).filter(
+        removeNullable
+      )
+      if (processedTracks.length === processedFiles.length) {
+        setUploadTrackError(null)
+      }
+
+      if (
+        uploadType === UploadType.INDIVIDUAL_TRACK &&
+        tracks.length + processedTracks.length > 1
+      ) {
+        setUploadType(UploadType.INDIVIDUAL_TRACKS)
+      }
+
+      setTracks([...tracks, ...processedTracks])
+    },
+    [tracks, uploadType]
+  )
+
+  const onRemoveTrack = useCallback(
+    (index: number) => {
+      setTracks(tracks.filter((_, i) => i !== index))
+      setUploadType(
+        tracks.length === 2 ? UploadType.INDIVIDUAL_TRACK : uploadType
+      )
+    },
+    [tracks, uploadType]
+  )
+
+  const onCloseMultiTrackNotification = useCallback(() => {
+    dispatch(toggleMultiTrackNotification(false))
+  }, [dispatch])
+
+  const stopPreview = useCallback(() => {
+    if (preview) {
+      preview.pause()
+      preview.currentTime = 0
+    }
+    setPreview(null)
+    setPreviewIndex(-1)
+  }, [preview])
+
+  const playPreview = useCallback(
+    (index: number) => {
+      // Stop existing music if some is playing.
+      if (playing) {
+        dispatch(pause())
+      }
+
+      if (preview) stopPreview()
+      const audio = tracks[index].preview
+      audio.play()
+      setPreview(audio)
+      setPreviewIndex(index)
+    },
+    [dispatch, playing, preview, stopPreview, tracks]
+  )
+
   return (
     <Page
       title='Upload'
       description='Upload and publish audio content to the Audius platform'
       contentClassName={styles.upload}
       header={<Header primary={'Upload'} />}
     >
-      This is the new upload page
+      <SelectPage
```

Comment:
sweet, up to you but i would to move all these props down to the leaf compnonets eventually. what do you think?

Reply:
Yes, I plan to. Just wanted to get a piece working for now 😄

Reply:
It wasn't a class component yet so I'll port the SelectPage and push the hooks, state, and callbacks down. I think in the end we'll want the shared page, form, and track state up here and the rest a level down

Reply:
dope thanks for context!